Psy 6 create a clear schedule button on homepage

Sam41704 opened the pull request:
Ready for Review. 

Delete schedule button added and tested. Also added ConfirmDialog.tsx with a reusable alert dialogue component.

Sam41704 committed:
Added delete schedule button to admin page

Added shadcn alert dialog package
Added button in AdminPage below "generate schedule" to "delete schedule", which puts in a request to  DELETE api/schedules/{id}.
Use shadcn alert dialog to confirm deletion.

Have not tested

Abhinav-Pollepalli committed:
Initial Docker setup

Abhinav-Pollepalli committed:
Docker setup and project files

Abhinav-Pollepalli committed:
Update README.md

Sam41704 committed:
Merge branch 'main' into PSY-6-Create-a-Clear-Schedule-Button-on-Homepage

Abhinav-Pollepalli committed:
Add encrypted database dump and update docker-compose file

Abhinav-Pollepalli committed:
Remove old db and init folders from branch

Abhinav-Pollepalli committed:
Updated README.md

Update README with latest setup instructions

Abhinav-Pollepalli committed:
Updated docker-compose.yml

Changed the port of the db

Abhinav-Pollepalli committed:
Merge remote-tracking branch 'origin/main' into PSY-8-Create-a-Scaffold-SQL-script-to-set-up-development-databases

Abhinav-Pollepalli committed:
upgraded to .NET 8 and resolved backend build issues

upgraded to .NET 8 and resolved backend build issues. Modified README.md

Abhinav-Pollepalli committed:
Update README.md

Abhinav-Pollepalli committed:
Update README.md

Update README.md

Abhinav-Pollepalli committed:
Update README.md

Abhinav-Pollepalli committed:
Fix .editorconfig formatting

Abhinav-Pollepalli committed:
Fixed formatting and whitespace issues .editorconfig

Fixed formatting and whitespace issues .editorconfig

Sam41704 committed:
Merge pull request #2 from lbrown169/PSY-8-Create-a-Scaffold-SQL-script-to-set-up-development-databases

Psy 8 create a scaffold sql script to set up development databases

Abhinav-Pollepalli committed:
replaced encrypted sql dump with fake sql dump

replaced encrypted sql dump with fake sql dump

Abhinav-Pollepalli committed:
Updated README.md

Updated README.md

Abhinav-Pollepalli committed:
Updated README.md

Updated README.md

Abhinav-Pollepalli committed:
Updated db connection string

Updated db connection string

Sam41704 committed:
Merge pull request #3 from lbrown169/PSY-8-Create-a-Scaffold-SQL-script-to-set-up-development-databases

Psy 8 create a scaffold sql script to set up development databases

Sam41704 committed:
Update locked_psycall_prod_dump.sql.enc

Sam41704 committed:
Merge branch 'PSY-6-Create-a-Clear-Schedule-Button-on-Homepage' of https://github.com/lbrown169/Medical-Resident-Scheduling into PSY-6-Create-a-Clear-Schedule-Button-on-Homepage

Sam41704 committed:
Delete Schedule working. Confirm Dialog component.

Moved Deletion alert dialogue into reusable component for future alert dialogues. see "ConfirmDialogue.tsx". Added a helper function to get the schedule id before passing to delete id endpoint

Sam41704 committed:
merge main into current branch

aitanoffir reviewed (changes requested):
in line 122 of AdminPage.tsx 
"const [deleteOpen, setDeleteOpen] = useState(false);"
These values seem to be unused and are causing an NPM build issue from an explicit type rule in Typescript. Either get rid of them or ensure they are used.

Also in ConfirmDialog.tsx there is a large block of commented out code. If this is needed in the future you can keep it if not please delete it.

Sam41704 committed:
Completed requested changes

Removed unused constant (forgot to delete earlier.

Removed commented notes in ConfirmDialog.tsx

Sam41704 committed:
Update AdminPage.tsx

Sam41704 committed:
Update formatting

Sam41704 commented:
I noticed that onNavigateToCalendar may not be working (at least on my end). It's supposed to redirect the user to the calendar after generating/deleting a schedule. Will add to list of bugs

aitanoffir reviewed (changes requested):
When deleting a schedule it does not redirect to the calendar and it does not re-render the calender so it still shows the old schedule. 

It seems like the "onNavigateToCalendar" is not actually defined in page.tsx and that is causing the issue

aitanoffir commented:
> I noticed that onNavigateToCalendar may not be working (at least on my end). It's supposed to redirect the user to the calendar after generating/deleting a schedule. Will add to list of bugs

I can make the fix directly in this branch and if it works approve the pull request

Commit:
Fixed redirect to calendar issue when generating/deleting schedule and deleted unnecessary timeouts

aitanoffir reviewed (approved):
tested and working
